### DISCUSSION TITLE
Getting the solution data of a reduced matrix in Maxwell

### DISCUSSION THREAD

**Discussion by AlfVII** [ansys/pyaedt#4790] [Q&A] [answered]
Hi!

I am creating a reduced matrix joining in parallel some windings, 
![image](https://github.com/ansys/pyaedt/assets/1822593/6fd23b41-64b6-47e9-b853-9057a4fe1b97)

and I want to get the solution data for that reduced matrix in Maxwell.
![image](https://github.com/ansys/pyaedt/assets/1822593/e03ee75d-e348-46e6-9952-d30b722c28a3)

But when I  try to get the solution data with pyaedt I have no way of selecting the reduced matrix.

I have tried setting getting the quantities with this:
`project.post.available_report_quantities(context="windings", quantities_category='L')`

but I am getting the ones from the Original matrix:
`['solution_matrix.L(Primary_winding_parallel_0,Primary_winding_parallel_0)', 'solution_matrix.L(Primary_winding_parallel_0,Primary_winding_parallel_1)', 'solution_matrix.L(Primary_winding_parallel_0,Secondary_winding_parallel_0)', 'solution_matrix.L(Primary_winding_parallel_0,Secondary_winding_parallel_1)', 'solution_matrix.L(Primary_winding_parallel_1,Primary_winding_parallel_0)', 'solution_matrix.L(Primary_winding_parallel_1,Primary_winding_parallel_1)', 'solution_matrix.L(Primary_winding_parallel_1,Secondary_winding_parallel_0)', 'solution_matrix.L(Primary_winding_parallel_1,Secondary_winding_parallel_1)', 'solution_matrix.L(Secondary_winding_parallel_0,Primary_winding_parallel_0)', 'solution_matrix.L(Secondary_winding_parallel_0,Primary_winding_parallel_1)', 'solution_matrix.L(Secondary_winding_parallel_0,Secondary_winding_parallel_0)', 'solution_matrix.L(Secondary_winding_parallel_0,Secondary_winding_parallel_1)', 'solution_matrix.L(Secondary_winding_parallel_1,Primary_winding_parallel_0)', 'solution_matrix.L(Secondary_winding_parallel_1,Primary_winding_parallel_1)', 'solution_matrix.L(Secondary_winding_parallel_1,Secondary_winding_parallel_0)', 'solution_matrix.L(Secondary_winding_parallel_1,Secondary_winding_parallel_1)']`

if I try to get the solution data with this:
`solution_data = self.project.post.get_solution_data("current_matrix.L(Primary,Secondary)", context="windings")`

I am getting:
```
PyAEDT INFO: Solution Data Correctly Loaded.
({'current_matrix.L(Primary,Secondary)': {(0.0001,): nan}}, {'current_matrix.L(Primary,Secondary)': {(0.0001,): 0}})
```

How can I get that solution data with Python?

Thanks!

**Comment by gmalinve** [comment, marked as answer]
Hi @AlfVII,

Thanks for opening this discussion!
We need to further investigate how to use _get_solution_data_ for reduced matrices. I'll open an issue for that and keep you posted.
For the moment, as a shortcut, I'd suggest to use export_rl_matrix() that allows you to export the reduced matrix in a txt file.
For example: 
`m2d.export_rl_matrix(matrix_name="ReduceMatrix1", output_file=r"C:\Test\reduced.txt")`

Hope this makes sense,

Kind regards.

Giulia
``
